src/lightning/pytorch/trainer/connectors/data_connector.py

```diff
@@ -333,83 +318,80 @@ def _reset_eval_dataloader(
         Returns:
             Tuple (num_batches, dataloaders)
         """
-        # always get the loaders first so we can count how many there are
         dataloaders = self._request_dataloader()
+        if not isinstance(dataloaders, CombinedLoader):
+            combined_loader = CombinedLoader(dataloaders, "sequential")
+        else:
+            combined_loader = dataloaders
 
-        if self.trainer.overfit_batches > 0:
-            dataloaders = self._resolve_overfit_batches(dataloaders, mode)
+        trainer = self.trainer
 
-        # TODO(carmocca): list conversion shouldn't be forced
-        if not isinstance(dataloaders, list):
-            dataloaders = [dataloaders]  # type: ignore[assignment]
+        if trainer.overfit_batches > 0:
+            self._resolve_overfit_batches(combined_loader, mode)
 
-        if any(dl is None for dl in dataloaders):
-            rank_zero_warn("One of given dataloaders is None and it will be skipped.")
```

Comment:
I didn't port this skip logic. I think it can lead to bugs where the user accidentally returns `None` and then training continues normally when it should fail.